modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java

```diff
@@ -919,13 +920,18 @@ private boolean _isSkipRedirect(HttpServletRequest httpServletRequest) {
 	}
 
 	private String _normalizeFriendlyURL(String friendlyURL) {
-		if (Validator.isNotNull(friendlyURL) &&
-			friendlyURL.startsWith(StringPool.SLASH)) {
+		String decodedFriendlyURL = HttpComponentsUtil.decodeURL(friendlyURL);
 
-			return friendlyURL.substring(1);
+		String normalizedFriendlyURL =
+			FriendlyURLNormalizerUtil.normalizeWithEncoding(decodedFriendlyURL);
+
+		if (Validator.isNotNull(normalizedFriendlyURL) &&
```

Comment:
I know that both methods check the null value, but maybe it's clearer to check the null before to decode and normalize the friendlyURL